### PR TITLE
wip

### PR DESCRIPTION
**Pull request by jonstorer**

### CODE REVIEW

File: spec/support/fixtures/repository.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (jonstorer) [inline spec/support/fixtures/repository.json]: rm

File: spec/support/fixtures/transaction.json [hunk 2]
```diff
@@ -0,0 +1,148 @@
+{
```
Comment (jonstorer) [inline spec/support/fixtures/transaction.json]: reset head

File: lib/vindicia.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+require 'httparty'
+require 'hashie'
+
+require 'vindicia/repository/base'
+require 'vindicia/request_mapper/base'
+require 'vindicia/response_mapper/base'
+require 'vindicia/model/base'
+#require 'vindicia/exception'
+
+module Vindicia
```
Comment (jonstorer) [inline lib/vindicia.rb]: redo this.

File: lib/vindicia.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+require 'httparty'
+require 'hashie'
+
+require 'vindicia/type'
+require 'vindicia/model'
+require 'vindicia/repository'
+require 'vindicia/request'
+require 'vindicia/response'
+
+module Vindicia
```
Comment (jonstorer) [inline lib/vindicia.rb]: this needs to be better

**Comment by douglasrlee** [comment]
Can you remove and ignore this: `lib/.DS_Store`

File: lib/vindicia.rb (hunk 4 above)
Comment (douglasrlee) [inline lib/vindicia.rb]: Better how?

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+sudo: false
+language: ruby
+rvm:
+  - 2.3.3
+before_install: gem install bundler -v 1.13.6
```
Comment (kylessnell) [inline .travis.yml]: travis?

File: CODE_OF_CONDUCT.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+faith may face temporary or permanent repercussions as determined by other
+members of the project's leadership.
+
+## Attribution
+
+This Code of Conduct is adapted from the [Contributor Covenant][homepage], version 1.4,
+available at [http://contributor-covenant.org/version/1/4][version]
+
+[homepage]: http://contributor-covenant.org
+[version]: http://contributor-covenant.org/version/1/4/
```
Comment (kylessnell) [inline CODE_OF_CONDUCT.md]: 👍

File: lib/vindicia/repository/base.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    def self.first
+      Vindicia::Response.new(Vindicia::Request.new(:get, route, { query: { limit: 1 } })).body.first
+    end
+
+    def self.find(id)
+      raise ArgumentError.new("Cannot find Resource with id 'nil'") unless id
+      Vindicia::Response.new(Vindicia::Request.new(:get, route(id)))
+    end
+
+    #def save(model)
```
Comment (douglasrlee) [inline lib/vindicia/repository/base.rb]: rm?

File: lib/vindicia/request.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
+module Vindicia
+  class Request
+    include HTTParty
+    #debug_output $stderr
```
Comment (douglasrlee) [inline lib/vindicia/request.rb]: rm?

File: lib/vindicia/model/price.rb [hunk 9]
```diff
@@ -0,0 +1,6 @@
+module Vindicia::Model
+  class Price < Base
+    property :amount, coerce: Float
```
Comment (kylessnell) [inline lib/vindicia/model/price.rb]: why not BigDecimal

File: lib/vindicia/model/product_price.rb [hunk 10]
```diff
@@ -0,0 +1,4 @@
+module Vindicia::Model
+  class ProductPrice < Vindicia::Model::Price
```
Comment (kylessnell) [inline lib/vindicia/model/product_price.rb]: why is this its own model

File: lib/vindicia/model/product_description.rb [hunk 11]
```diff
@@ -0,0 +1,6 @@
+module Vindicia::Model
+  class ProductDescription < Vindicia::Model::Description
```
Comment (kylessnell) [inline lib/vindicia/model/product_description.rb]: why is this its own model

File: lib/vindicia/model/price.rb (hunk 9 above)
Comment (kylessnell) [inline lib/vindicia/model/price.rb]: and why is Price its own model

File: .travis.yml (hunk 5 above)
Comment (jonstorer) [inline .travis.yml]: i used the bundler gem builder and this came with, I haven't removed it or decided to use travis

File: CODE_OF_CONDUCT.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
+# Contributor Covenant Code of Conduct
```
Comment (jonstorer) [inline CODE_OF_CONDUCT.md]: this came with the gem builder.. it's like `npm init` and it asks a bunch of questions

File: CODE_OF_CONDUCT.md (hunk 6 above)
Comment (jonstorer) [inline CODE_OF_CONDUCT.md]: tbh, i have no idea what this means, but I'm all about open standards

File: lib/vindicia.rb (hunk 4 above)
Comment (jonstorer) [inline lib/vindicia.rb]: the way to configure this app right now is really slow an sloppy, I want to clean up this top level configurer

File: lib/vindicia/model/price.rb (hunk 9 above)
Comment (jonstorer) [inline lib/vindicia/model/price.rb]: the models in this lib are based on responses I've received from their API. At one point, I must have received a price object.

File: lib/vindicia/model/product_description.rb (hunk 11 above)
Comment (jonstorer) [inline lib/vindicia/model/product_description.rb]: their API asks for it link this

File: lib/vindicia/model/product_price.rb (hunk 10 above)
Comment (jonstorer) [inline lib/vindicia/model/product_price.rb]: their API asks for it link this

File: lib/vindicia/repository/base.rb (hunk 7 above)
Comment (jonstorer) [inline lib/vindicia/repository/base.rb]: yup

File: lib/vindicia/repository/billing_plan.rb [hunk 13]
```diff
@@ -0,0 +1,5 @@
+module Vindicia::Repository
+  class BillingPlan < Base
+  end
+end
+
```
Comment (jonstorer) [inline lib/vindicia/repository/billing_plan.rb]: ws

File: lib/vindicia/request.rb (hunk 8 above)
Comment (jonstorer) [inline lib/vindicia/request.rb]: yeah, I'll find a way to configure this from the top module

File: lib/vindicia/request_mapper/base.rb [hunk 14]
```diff
@@ -0,0 +1,31 @@
+module Vindicia::RequestMapper
```
Comment (jonstorer) [inline lib/vindicia/request_mapper/base.rb]: I can remove all request mappers right now,,, they're not used

File: lib/vindicia/response.rb [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+    private
+
+    def map(hash)
+      case hash['object']
+      when 'List'
+        hash['data'].map {|d| map(d) }
+      when 'Error'
+        throw Vindicia::Exception.new(response.to_h)
+      else
+        "Vindicia::Model::#{hash['object']}".constantize.new(hash)
```
Comment (jonstorer) [inline lib/vindicia/response.rb]: this line here is why I need to define every object they send back.

File: spec/unit/model/base_spec.rb [hunk 16]
```diff
@@ -0,0 +1,17 @@
+require 'spec_helper'
+
+describe Vindicia::Model::Base do
+  it { should be_a(Hashie::Trash) }
+  it { should be_a(Hashie::Extensions::MergeInitializer) }
+  it { should be_a(Hashie::Extensions::IgnoreUndeclared) }
+  it { should be_a(Hashie::Extensions::IndifferentAccess) }
+  it { should be_a(Hashie::Extensions::Dash::Coercion) }
+
+  context '#object' do
```
Comment (jonstorer) [inline spec/unit/model/base_spec.rb]: switch to `its`

File: spec/unit/repository_spec.rb [hunk 17]
```diff
@@ -0,0 +1,215 @@
+#allow(Vindicia::Request).to receive(:new).with(:get, '/bases/1').and_return(request)
```
Comment (jonstorer) [inline spec/unit/repository_spec.rb]: rm

File: spec/unit/repository_spec.rb [hunk 18]
```diff
@@ -0,0 +1,215 @@
...
+      it 'returns the correct response' do
+        expect(call_response).to eql(record)
+      end
+
+      it 'sets limit to 1' do
+        expect(Vindicia::Request).to have_received(:new).with(anything, anything, { query: { limit: 1 } })
+      end
+    end
+
+    #describe '#save' do
```
Comment (jonstorer) [inline spec/unit/repository_spec.rb]: rm

File: lib/vindicia/response.rb (hunk 15 above)
Comment (douglasrlee) [inline lib/vindicia/response.rb]: And if not would it just be raw?

File: spec/integration/transaction_spec.rb [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+
+    its(:vid)          { is_expected.to eql('00df553ba17a0cde2275e278628fd6b7f2d8b067') }
+    its(:id)           { is_expected.to eql('LEGALSHI00000684') }
+    its(:created)      { is_expected.to eql(DateTime.parse('2017-02-11T00:01:35-08:00')) }
+    its(:amount)       { is_expected.to eql(10.71) }
+    its(:currency)     { is_expected.to eql('USD') }
+    its(:account)      { is_expected.to be_a(Vindicia::Model::Account)  }
+    its(:subscription) { is_expected.to be_a(Vindicia::Model::Subscription)  }
+
+    #its(:descriptions)         { is_expected.to be_an(Array)  }
```
Comment (douglasrlee) [inline spec/integration/transaction_spec.rb]: rm?

File: lib/vindicia/repository/base.rb [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+      #Vindicia::Response.new(Vindicia::Request.new(:post, route(id), Vindicia::RequestMapper::Base.map(model)))
+    #end
```
Comment (kylessnell) [inline lib/vindicia/repository/base.rb]: 💀

File: lib/vindicia/repository/base.rb [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+
+    private
+
+    def self.route(id=nil)
+      class_name = self.name.split('::').last.downcase
+      plural_class_name = ActiveSupport::Inflector.pluralize(class_name)
+      [ '', plural_class_name, id ].compact.join('/')
```
Comment (kylessnell) [inline lib/vindicia/repository/base.rb]: why do you need the empty sting

File: lib/vindicia/request.rb (hunk 8 above)
Comment (kylessnell) [inline lib/vindicia/request.rb]: 💀

File: lib/vindicia/request_mapper/base.rb [hunk 22]
```diff
@@ -0,0 +1,31 @@
...
+  class Base
+    extend Forwardable
+    def_delegators :@instance, :object
+
+    def initialize(instance)
+      @instance = instance
```
Comment (kylessnell) [inline lib/vindicia/request_mapper/base.rb]: where are you using this instance var?

File: lib/vindicia/request_mapper/base.rb [hunk 23]
```diff
@@ -0,0 +1,31 @@
...
+    end
+
+    def to_json(*options)
+      as_json(*options).to_json(*options)
+    end
+
+    def self.map(instance)
+      "Vindicia::RequestMapper::#{instance.object}".constantize.new(instance)
```
Comment (kylessnell) [inline lib/vindicia/request_mapper/base.rb]: use instance var here?

File: lib/vindicia/request_mapper/base.rb [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+    end
+
+    private
+
+    def as_json(options={})
```
Comment (kylessnell) [inline lib/vindicia/request_mapper/base.rb]: lotta monkey patching

File: lib/vindicia/request_mapper/base.rb [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+      (self.public_methods - Object.methods).inject({}) do |memo, method_name|
+        memo[method_name] = self.send(method_name)
+        memo
+      end
+    end
+
+    #def _present(*args)
+      #self.class._present(*args)
+    #end
```
Comment (kylessnell) [inline lib/vindicia/request_mapper/base.rb]: 💀

File: lib/vindicia/request_mapper/product.rb [hunk 26]
```diff
@@ -0,0 +1,15 @@
+module Vindicia::RequestMapper
+  class Product < Base
+    def_delegators :@model, :id
+
+    def default_billing_plan
+      _present(@model.default_billing_plan) if @model.default_billing_plan
```
Comment (kylessnell) [inline lib/vindicia/request_mapper/product.rb]: the `_present` method is commented out above

File: lib/vindicia/type/boolean.rb [hunk 27]
```diff
@@ -0,0 +1,14 @@
+module Vindicia::Type
+  class Boolean
+    def self.coerce(value)
+      case value
+      when String
+        !!(value =~ /\A(true|t|yes|y|1)\z/i)
+      when Numeric
+        !value.to_i.zero?
+      else
+        value == true
```
Comment (kylessnell) [inline lib/vindicia/type/boolean.rb]: `value`

File: spec/integration/product_spec.rb [hunk 28]
```diff
@@ -0,0 +1,42 @@
...
+
+  describe 'first products' do
+    subject do
+      Vindicia::Repository::Product.first
+    end
+
+    before do
+      stub_get('/products')
+        .with({ query: { limit: 1 } })
+        .to_return(:status => 200, :body => fixture('product'), :headers => { 'Content-Type': 'application/json' })
```
Comment (kylessnell) [inline spec/integration/product_spec.rb]: multiline this plz

File: spec/integration/transaction_spec.rb [hunk 29]
```diff
@@ -0,0 +1,46 @@
...
+      #expect(subject.descriptions.first).to be_a(Vindicia::Model::ProductDescription)
+    #end
+
+    #it 'entitlements are Vindicia::Model::Entitlement' do
+      #expect(subject.entitlements.first).to be_a(Vindicia::Model::Entitlement)
+    #end
+
+    #it 'product prices are Vindicia::Model::ProductPrice' do
+      #expect(subject.prices.first).to be_a(Vindicia::Model::ProductPrice)
+    #end
```
Comment (kylessnell) [inline spec/integration/transaction_spec.rb]: so many comments

File: spec/support/fixtures/product.json [hunk 30]
```diff
@@ -0,0 +1,75 @@
...
+      "id": "1",
+      "vid": "7118699d6406494d7672503761fa21809fefaf25",
+      "created": "2017-02-03T13:46:27-08:00",
+      "descriptions": {
+        "object": "List",
+        "data": [
+          {
+            "object": "ProductDescription",
+            "language": "EN",
+            "description": "Oklahoma Limited Personal Plan"
```
Comment (kylessnell) [inline spec/support/fixtures/product.json]: legalshield specific

File: spec/support/fixtures/transaction.json [hunk 31]
```diff
@@ -0,0 +1,148 @@
...
+  "object": "Transaction",
+  "id": "LEGALSHI00000597",
```
Comment (kylessnell) [inline spec/support/fixtures/transaction.json]: ls specific

File: spec/support/fixtures/transaction.json [hunk 32]
```diff
@@ -0,0 +1,148 @@
...
+      {
+        "object": "TransactionStatus",
+        "status": "AuthorizationPending",
+        "created": "2017-02-10T13:46:52-08:00",
+        "payment_method_type": "PayPal",
+        "paypal_status": {
+          "object": "TransactionStatusPayPal",
+          "token": "EC-0PY59464AS577883J",
+          "auth_code": "000",
+          "redirect_url": "https://www.sandbox.paypal.com/webscr?cmd=_express-checkout&useraction=commit&token=EC-0PY59464AS577883J"
```
Comment (kylessnell) [inline spec/support/fixtures/transaction.json]: probs shouldn't commit this

File: spec/support/fixtures/transaction.json [hunk 33]
```diff
@@ -0,0 +1,148 @@
...
+      {
+        "object": "TransactionStatus",
+        "status": "New",
+        "created": "2017-02-10T13:46:50-08:00",
+        "payment_method_type": "PayPal",
+        "paypal_status": {
+          "object": "TransactionStatusPayPal",
+          "token": "EC-0PY59464AS577883J",
+          "auth_code": "000",
+          "redirect_url": "https://www.sandbox.paypal.com/webscr?cmd=_express-checkout&useraction=commit&token=EC-0PY59464AS577883J"
```
Comment (kylessnell) [inline spec/support/fixtures/transaction.json]: same

File: spec/support/fixtures/transaction.json [hunk 34]
```diff
@@ -0,0 +1,148 @@
...
+        }
+      }
+    ],
+    "total_count": 2
+  },
+  "payment_processor": "PayPal",
+  "payment_processor_transaction_id": "LEGALSHI00000597",
```
Comment (kylessnell) [inline spec/support/fixtures/transaction.json]: ls specific

File: spec/support/fixtures/transaction_status_paypal.json [hunk 35]
```diff
@@ -0,0 +1,6 @@
+{
+  "object": "TransactionStatusPayPal",
+  "token": "EC-0PY59464AS577883J",
+  "auth_code": "000",
+  "redirect_url": "https://www.sandbox.paypal.com/webscr?cmd=_express-checkout&useraction=commit&token=EC-0PY59464AS577883J"
```
Comment (kylessnell) [inline spec/support/fixtures/transaction_status_paypal.json]: don't commit

File: spec/support/fixtures/transactions.json [hunk 36]
```diff
@@ -0,0 +1,828 @@
...
+              "customer_specified_type": "Visa",
+              "customer_description": "My wifes Visa credit card",
+              "primary": false,
+              "active": true
+            }
+          ],
+          "total_count": 2
+        },
+        "metadata": {
+          "Created By": "Vindicia sandbox software"
```
Comment (kylessnell) [inline spec/support/fixtures/transactions.json]: ?

File: spec/support/fixtures/transactions.json [hunk 37]
```diff
@@ -0,0 +1,828 @@
...
+          {
+            "object": "TransactionStatus",
+            "status": "Cancelled",
+            "created": "2017-02-11T00:10:55-08:00",
+            "payment_method_type": "PayPal",
+            "paypal_status": {
+              "object": "TransactionStatusPayPal",
+              "token": "EC-64F78649YT8003602",
+              "auth_code": "11451",
+              "redirect_url": "https://www.sandbox.paypal.com/webscr?cmd=_express-checkout&useraction=commit&token=EC-64F78649YT8003602"
```
Comment (kylessnell) [inline spec/support/fixtures/transactions.json]: don't commit

File: spec/support/fixtures/transactions.json [hunk 38]
```diff
@@ -0,0 +1,828 @@
...
+          {
+            "object": "TransactionStatus",
+            "status": "New",
+            "created": "2017-02-11T00:01:35-08:00",
+            "payment_method_type": "PayPal",
+            "paypal_status": {
+              "object": "TransactionStatusPayPal",
+              "token": "EC-64F78649YT8003602",
+              "auth_code": "11451",
+              "redirect_url": "https://www.sandbox.paypal.com/webscr?cmd=_express-checkout&useraction=commit&token=EC-64F78649YT8003602"
```
Comment (kylessnell) [inline spec/support/fixtures/transactions.json]: same

File: spec/support/fixtures/transactions.json [hunk 39]
```diff
@@ -0,0 +1,828 @@
...
+            "service_period_ends": "2017-02-11T00:00:00-08:00",
+            "tax_type": "Exclusive Sales",
+            "total": 0
+          },
+          {
+            "object": "TransactionItem",
+            "sku": "1",
+            "index_number": 2,
+            "item_type": "Purchase",
+            "name": "Oklahoma Limited Personal Plan",
```
Comment (kylessnell) [inline spec/support/fixtures/transactions.json]: ls specific

File: spec/support/fixtures/transactions.json [hunk 40]
```diff
@@ -0,0 +1,828 @@
...
+          {
+            "object": "TransactionStatus",
+            "status": "AuthorizationPending",
+            "created": "2017-02-10T13:19:05-08:00",
+            "payment_method_type": "PayPal",
+            "paypal_status": {
+              "object": "TransactionStatusPayPal",
+              "token": "EC-64F78649YT8003602",
+              "auth_code": "000",
+              "redirect_url": "https://www.sandbox.paypal.com/webscr?cmd=_express-checkout&useraction=commit&token=EC-64F78649YT8003602"
```
Comment (kylessnell) [inline spec/support/fixtures/transactions.json]: same

File: spec/support/fixtures/transactions.json [hunk 41]
```diff
@@ -0,0 +1,828 @@
...
+          {
+            "object": "TransactionStatus",
+            "status": "New",
+            "created": "2017-02-10T13:19:03-08:00",
+            "payment_method_type": "PayPal",
+            "paypal_status": {
+              "object": "TransactionStatusPayPal",
+              "token": "EC-64F78649YT8003602",
+              "auth_code": "000",
+              "redirect_url": "https://www.sandbox.paypal.com/webscr?cmd=_express-checkout&useraction=commit&token=EC-64F78649YT8003602"
```
Comment (kylessnell) [inline spec/support/fixtures/transactions.json]: saaaaaaaaaaaaaaaaame

File: spec/support/fixtures/transactions.json [hunk 42]
```diff
@@ -0,0 +1,828 @@
...
+            }
+          }
+        ],
+        "total_count": 2
+      },
+      "payment_processor": "PayPal",
+      "payment_processor_transaction_id": "LEGALSHI00000591",
```
Comment (kylessnell) [inline spec/support/fixtures/transactions.json]: LS specific

File: spec/support/fixtures/transactions.json [hunk 43]
```diff
@@ -0,0 +1,828 @@
...
+            "service_period_ends": "2017-02-10T00:00:00-08:00",
+            "tax_type": "Exclusive Sales",
+            "total": 0
+          },
+          {
+            "object": "TransactionItem",
+            "sku": "1",
+            "index_number": 2,
+            "item_type": "Purchase",
+            "name": "Oklahoma Limited Personal Plan",
```
Comment (kylessnell) [inline spec/support/fixtures/transactions.json]: LS specific

File: spec/support/fixtures/transactions.json [hunk 44]
```diff
@@ -0,0 +1,828 @@
...
+              "customer_specified_type": "Visa",
+              "customer_description": "My wifes Visa credit card",
+              "primary": false,
+              "active": true
+            }
+          ],
+          "total_count": 2
+        },
+        "metadata": {
+          "Created By": "Vindicia sandbox software"
```
Comment (kylessnell) [inline spec/support/fixtures/transactions.json]: ?

File: spec/support/fixtures/transactions.json [hunk 45]
```diff
@@ -0,0 +1,828 @@
...
+        "line1": "4635 Ingram Road",
+        "city": "Liberty",
+        "district": "NC",
+        "postal_code": "27298",
+        "country": "US",
+        "phone": "336-622-3355",
+        "fax": "752-331-6484"
+      },
+      "metadata": {
+        "Created By": "Vindicia sandbox software"
```
Comment (kylessnell) [inline spec/support/fixtures/transactions.json]: ?

File: spec/support/helpers/webmock.rb [hunk 46]
```diff
@@ -0,0 +1,17 @@
+require 'webmock/rspec'
+
+WebMock.disable_net_connect!
+
+FIXTURE_PATH = File.expand_path('../../fixtures', __FILE__)
+
+def stub_get(path)
+  stub_request(:get, Vindicia::Request.base_uri + path)
+end
```
Comment (kylessnell) [inline spec/support/helpers/webmock.rb]: why not use something like VCR?

File: spec/unit/model/account_spec.rb [hunk 47]
```diff
@@ -0,0 +1,19 @@
+require 'spec_helper'
```
Comment (kylessnell) [inline spec/unit/model/account_spec.rb]: your PR is way too long for me to check that all of these tests are giving you full coverage

File: spec/unit/repository_spec.rb [hunk 48]
```diff
@@ -0,0 +1,215 @@
+#allow(Vindicia::Request).to receive(:new).with(:get, '/bases/1').and_return(request)
+#allow(Vindicia::Request).to receive(:new).with(:get, '/bases').and_return(request)
+#allow(Vindicia::Request).to receive(:get).with(:get, '/bases', { query: { } }).and_return(request)
+#allow(Vindicia::Request).to receive(:get).with(:get, '/bases', { query: { limit: 1 } }).and_return(response)
+#allow(Vindicia::Request).to receive(:new).with(:get, '/bases', { query: { limit: 100 } }).and_return(request)
+#allow(Vindicia::Request).to receive(:new).with(:get, '/bases', { query: { limit: 100, name: 'Jon' } }).and_return(request)
```
Comment (kylessnell) [inline spec/unit/repository_spec.rb]: 💀 💀 💀

File: spec/unit/repository_spec.rb [hunk 49]
```diff
@@ -0,0 +1,215 @@
...
+
+        #it 'delegates response mapping correctly' do
+          #expect(Vindicia::Response).to have_received(:new).with(data)
+        #end
+
+        #it 'returns the mapped response' do
+          #expect(call_response).to eql(mapped_response)
+        #end
+      #end
+    #end
```
Comment (kylessnell) [inline spec/unit/repository_spec.rb]: [http://i0.kym-cdn.com/photos/images/masonry/000/259/943/694.png](url)

File: spec/unit/vindicia_spec.rb [hunk 50]
```diff
@@ -0,0 +1,22 @@
...
+    expect(Vindicia::VERSION).not_to be nil
+  end
+
+  it { should have_attr_accessor(:username) }
+  it { should have_attr_accessor(:password) }
+
+  it 'sets sandbox values' do
+    Vindicia.sandbox!
+    expect(Vindicia::Request.base_uri).to eql('https://api.prodtest.vindicia.com')
+  end
```
Comment (kylessnell) [inline spec/unit/vindicia_spec.rb]: why do we want this in the gem? or is this temp?

File: lib/vindicia/repository/base.rb (hunk 21 above)
Comment (jonstorer) [inline lib/vindicia/repository/base.rb]: to get the leading '/'

File: spec/support/fixtures/transactions.json (hunk 44 above)
Comment (jonstorer) [inline spec/support/fixtures/transactions.json]: no idea... it's just from their sandbox... I'll get to it understanding and incorporating it eventually

File: spec/unit/vindicia_spec.rb (hunk 50 above)
Comment (jonstorer) [inline spec/unit/vindicia_spec.rb]: @kylessnell the sidekiq get as a "test mode" option. I'm building the simplest version of that by changing the base url to example.com. I expect that the end, I'll have sandbox! and test! which will change the base url correctly, they're to help in testing

File: lib/vindicia/model/price.rb (hunk 9 above)
Comment (jonstorer) [inline lib/vindicia/model/price.rb]: added an issue

File: lib/vindicia/repository/base.rb (hunk 21 above)
Comment (jonstorer) [inline lib/vindicia/repository/base.rb]: i added an issue

File: spec/support/helpers/webmock.rb (hunk 46 above)
Comment (jonstorer) [inline spec/support/helpers/webmock.rb]: i added an issue

File: spec/unit/model/account_spec.rb (hunk 47 above)
Comment (jonstorer) [inline spec/unit/model/account_spec.rb]: i know, i'm working on closing this PR asap so I can do smaller ones

**Pull request merged**